### DISCUSSION TITLE
Design crashed after update and app import. No style options anymore

### DISCUSSION THREAD

**Discussion by sergey-dme** [Budibase/budibase#1888] [Help] [answered]
Just updated to v.0.9.67. with a new limitations of design tools... in new frontend you removed almost all useful style properties and design features for containers and other objects.

I spent two weeks creating UI in my app for few forms, navigation, cards, but now when I imported my app - there are no colors, borders, sizes, shadows, almost everything I did - lost. There's no ability to control anymore basic useful style properties using the builder interface/panels. 

Using custom CSS field/code for basic features/properties (that were in builder before the update) is a nightmare and CSS coding knowledge required.

**New way/limitation to use only custom CSS/code for basic important style properties is NO low-code at all and this is contrary to the low-code concept.**

Never been such disappointed before. 
That's really sad 😢

Just an example:

![01form](https://user-images.githubusercontent.com/53024832/124027232-4e620f00-d9fb-11eb-80e1-16631320e226.png)

Now I have no idea how to remove these new unwanted borders, how to restore font style, weight, colors, shadows... by manually coding CSS only?

I kindly ask you to return previous tools in the budibase builder 🙏 
The ability to control style / CSS properties was one of the main advantages of budibase ⚡️
Why you limited user by removing almost all style properties?

If it is impossible please advise how I can use/downgrade to v. 0.9.65 I would stay on this version forever as it has amazing tools to customize and design user' app.

Transitions style - also removed, these are very subjective things as the design in a whole, but it helps to bring "life" in user app UI and sometimes some boring forms can breathe with fly/fade/scale fx. 

Btw, we are living now not in the middle of 1990x with boring UIs with repetitive ugly design - the visual side is important nowadays and Java/HTML5/CSS3 give user incredible things for customization and design.
Please return it 🙏

**Comment by shogunpurple** [comment, marked as answer]
Hi @sergey-dme, first of all we are sorry to hear that you are disappointed with the latest updates to budibase. Firstly, I'd like to address the highest priority item - your app breaking after the update. There are actually a few bugs here around colors, opacity and the borders in your app. **We will have a fix shipped for these over the next few days**, which should fix your application and your styles.

> I kindly ask you to return previous tools in the budibase builder 🙏
The ability to control style / CSS properties was one of the main advantages of budibase ⚡️
Why you limited user by removing almost all style properties?

I'd like to clarify that we will not be returning the previous options to the budibase builder, but it is not a decision that we take lightly. The reason we have done this is that the current implementation does not fit the long term vision for the project. 

Budibase is built for speed, by creating beautiful apps quickly and easily by default.  We want to alleviate the stress, hassle and amount of decisions that our users have to make. Providing totally flexible design options for every single component makes for a busy UI, and adds a degree of complexity to budibase. There are **two major features** in the pipeline that we are currently working on, that make use of this new and simplified properties panel.

The first is conditional/bindable properties for styling/settings. This means you can use handlebars expressions in all of the properties for every component, giving you a much more interactive experience and will unlock use cases such as showing/hiding components, or changing colours based on data.

The second one that we are currently working on **full app theming**. This allows users to control CSS variables at the **app level** and to even apply some base themes, similar to how you can in your builder admin panel at the moment. This makes customising budibase apps much quicker and simpler. You won't have to spend 2 weeks styling and creating your app, because you'll be able to do it in a matter of minutes.

> New way/limitation to use only custom CSS/code for basic important style properties is NO low-code at all and this is contrary to the low-code concept.

By definition, low-code is a visual approach to software development. By abstracting and simplifying areas of our builder, we can help people create amazing applications with unrivalled simplicity. The idea is that you should not have to write a ton of CSS in budibase, because the customisation can be done with simple settings at the app level. We think you will really like this feature when it's shipped because it will allow you to add your own personal touch to your budibase apps.

Downgrading to 0.9.65 is also unfortunately not possible. It's also not advised as budibase is in beta and there are many new features and security updates that you will miss out on by not updating budibase.

We appreciate the time and effort that you put into this feedback and your other contributions to our community. Thank you for helping to make budibase better.

**Reply by sergey-dme** [reply]
@shogunpurple Thank you for the explanation, well I understand this is growing and continuously improving project, removing something for optimization purposes you will be implementing some new cool stuff. That's sound interesting. 
Looking forward to releasing a fix that will allow to retain user's design/style after app import which was made in previous version.

**Reply by Manishkparmar80** [reply]
Hi @sergey-dme , by looking at scrrenshot , i can see you have build Beautiful GUI, is it possible to provide exported app for self traning.

**Reply by sergey-dme** [reply]
@shogunpurple Whilst new functionality is not yet released, is there a way for users to create new page content or edit existing ones in the current version? 

For example, now for basic adjustments like changing containers sizes, colors, font attributes, my approach is quite complicated: when I create new or edit my existing app content - I open Google Chrome inspector, navigate to its css section  to check sizes, colors, fonts, other attributes then copy a css property I need or code part in the builder and edit it manually. 

For colors picker I use photoshop running just to pick and copy a color / HEX value. Old color picker integrated in builder was amazing.

This approach looks for me the only way for now to edit content, It is not fast, takes incomparable much more time then before for simple style adjustments. Otherwise builder in new version almost unusable for building content - no controls/options, just a few for alignment and margins...

**Reply by sergey-dme** [reply]
> Hi @sergey-dme , by looking at scrrenshot , i can see you have build Beautiful GUI, is it possible to provide exported app for self traning.

Thanks. Actually it's very simple and not yet finished, I will be providing exported app to you when finished and the fix for breaking design of imported apps released.

**Reply by shogunpurple** [reply]
@sergey-dme which styling options do you feel are missing from the container component?

**Reply by joebudi** [reply]
Hey @sergey-dme 

I am the co-founder of Budibase, and I take a prominent role in how the platform is designed. First of all, thanks for taking the time to share your experience. Your app looks great. I'd love to have a conversation with you to explore your needs, wants, and current problems. I would also like to explain why we made these changes and our long-term plan. Here is a link to book time in my calendar:
https://calendly.com/budibase

Speak soon, Joe.

**Reply by Manishkparmar80** [reply]
> > Hi @sergey-dme , by looking at scrrenshot , i can see you have build Beautiful GUI, is it possible to provide exported app for self traning.
> 
> Thanks. Actually it's very simple and not yet finished, I will be providing exported app to you when finished and the fix for breaking design of imported apps released.

Noted.  Tag me whenever you upload exported app.

**Reply by sergey-dme** [reply]
> @sergey-dme which styling options do you feel are missing from the container component?

Hi @shogunpurple I feel the following options are missing (most used before the update):

**Container component:**
- Layout (display: block/flex)
- Size (width/height, max width/height)
- Positions (static/relative/fixed/  
  absolute/sticky, Top, Bottom, L, R)
- Background Color/Image
- Border section
- Shadow section
- Transition style(fly, fade, scale)


**Headline, Paragraph:**
- Typography (font, font size, weight)
- Color


**Button**
- Typography + text color
- Bg color + Hover Bg color
- Border Radius, Color


**Link**
- Typography 
- Color + Hover Color 

**Icon**
- +Hover Color



**Header with Navigation section (in Layout):**

New one is cool, but it would be great to have an option with different header types depending on the logo position:
- with logo above the navigation links, the current high one - default one
- with logo on the same horizontal line as the navigation links, like mine: 

![image](https://user-images.githubusercontent.com/53024832/124211735-4dfa6e80-daf6-11eb-844a-27897e093173.jpeg)

My app is not finished. Before the update to v.0.9.67 I was halfway to finish it to start testing with my colleagues. Now, after the update finishing goes very slowly, as no style options in builder anymore and I use CSS coding and Chrome Inspector.

This header is totally custom one, done in old awesome builder. the logo is a svg code (inside url()) that I pasted in container background property. Added `cursor: pointer` in CSS as well and placed container with a link component to url "/" (home) and with empty link txt. All of that wasn't easy, but I did it thank to containers options :) 
The Log Out link in my header is a fake link, it doesn't work, I don't know how to get it working. The url like `/app/logout` or similar would be great to have, to use it on custom headers. The footer - is a common thing and also would be nice to have.

So, in your new header also would be nice to have options for logo, not only logo url to use external one, but also ability to paste svg code, options for logo size and positions also required as well as 'typography' and color (+hover) for navigation sections (links). 
I'd love to use your new header with awesome sticky and collapsing options, but it doesn't match my style, and has no option to control the style (height, logo position, links color+hover, even by using CSS it's impossible as links not displaying as child in the three anymore).

@shogunpurple @joebudi I chose budibase also because of the old builder with abilities to customize and control app style.

I'm sure many users would like to match their app style to organization brand style or create just their custom UI style for their apps. Visual part and UI are very important nowadays and style options are relevant. Giving freedom and tools to user to create and customize beautiful apps without coding and not limiting to only one "default" style is very relevant.

Thank you.

**Reply by sergey-dme** [reply]
> Hey @sergey-dme 
> 
> 
> 
> I am the co-founder of Budibase, and I take a prominent role in how the platform is designed. First of all, thanks for taking the time to share your experience. Your app looks great. I'd love to have a conversation with you to explore your needs, wants, and current problems. I would also like to explain why we made these changes and our long-term plan. Here is a link to book time in my calendar:
> 
> https://calendly.com/budibase
> 
> 
> 
> Speak soon, Joe.

Hey @joebudi Thank you for offering the conversation. I'd love to have it as well. I will booked time on next week I think. Looking forward to talking with you.

Thanks again.

Regards,
Sergey
